proposals/apple-mdm/Fleet-MDM-Plan.md

```diff
@@ -0,0 +1,218 @@
+
+Architecture of the current PoC:
+```mermaid
+flowchart LR
+
+subgraph Fleet [Fleet Server]
+
+    direction TB;
+
+    enroll[".mobileconfig<br>/mdm/apple/api/enroll"];
+    subgraph nanoMDMModules ["nanoMDM modules"]
+        direction TB;
+        nanoSCEP["nanoSCEP<br>/mdm/apple/scep"];
+        nanoMDM["nanoMDM<br>/mdm/apple/mdm"];
+        nanoDEP["nanoDEP<br>/mdm/apple/dep/proxy"];
+    end
+    MunkiRepo["Munki Repository<br>/mdm/apple/munki/repo"];
+    subgraph MunkiPkg [Munki Package Server];
+        manifest["manifest<br>/mdm/apple/munki/manifest"]
+        munkitools["munkitools-*.pkg<br>(signed)<br>/mdm/apple/munki/pkg"]
+    end
+    subgraph MySQL
+        direction LR;
+        mdmAppleDB[(nanoMDM<br>Schemas)];
+        fleetDB[(fleet)];
+    end
+end
+
+ApplePush[https://api.push.apple.com];
+AppleDEP[https://mdmenrollment.apple.com];
+nanoDEP -- Apple MDM DEP API ----> AppleDEP;
+nanoMDM --> ApplePush;
+
+nanoDEP --> mdmAppleDB;
+nanoMDM --> mdmAppleDB;
+```
+
+## Plan for Dogfood-MVP
+
+### Setup
+
+See [Fleet MDM setup](./Fleet-MDM-Setup.md).
+
+### Authority
+
+Only global admins will be allowed to use MDM functionality.
+
+### MDM features
+
+#### Enrollments
+
+"Enrollments" hold some settings for devices that will be enrolled to MDM.
+The MDM "enrollments" will allow Fleet to automatically enroll devices to specific teams, which then allows for applying specific MDM settings (depending on the team).
+
+For Dogfood-MVP, Fleet will allow creating global enrollments only (team support will be added at a subsequent iteration).
+Users will be able to create the two following types of enrollments:
+- Global manual enrollment
+- Global DEP enrollment
+
+We'll have a new `apple_manual_enrollments` table with the following fields:
+- ID (to deduce an "Enroll URL")
+- Name
+- Configuration (JSON?)
+- Team (or global)
+- Authentication configuration (initially NULL, meaning not authenticated)
+
+We'll have a new `apple_dep_enrollments` table with the following fields:
+- ID
+- Name
+- Configuration (JSON?)
+- Team (or global)
+- Authentication configuration (initially NULL, meaning not authenticated)
+- Reference to `dep_names` table (will be NULL at first)
+
+Fleetctl commands (and APIs):
+- Create enrollments:
+	`fleetctl apple-mdm enrollments create --configuration=<TBD> --name=Foo`
+
+- List enrollments (the "global" manual enroll and the DEP enroll):
+	`fleetctl apple-mdm enrollments list`
+
+##### DEP enrollment setup
+
+These commands use the <ENROLLMENT_ID>, which is the `enrollments.ID` (obtained via `fleetctl apple-mdm enrollments list`).
+The token downloaded from https://business.apple.com basically links a "DEP enrollment" in Fleet (entry in `apple_dep_enrollments`) to what Apple calls an "MDM server" in https://business.apple.com.
+
+###### 1. `fleetctl apple-mdm enrollments dep set-token setup --enrollment=<ENROLLMENT_ID>`
+
+- Checks the `ENROLLMENT_ID` is a DEP enrollment.
```

Comment:
By asking apple, or it can be done locally?

Reply:
locally.